src/mod_mam_sql.erl

```diff
@@ -514,6 +605,14 @@ get_max_direction_id(RSM) ->
 	    {undefined, undefined, <<>>}
     end.
 
+get_thread(RSM) ->
```

Comment:
This is where the hack-magic happens. We check the `<index>` element of the RSM query, and if it's present and an integer, we return a value. Otherwise we return `undefined`. This allows us to determine whether the client wants to get information from a thread or not.